gpt_index/vector_stores/deeplake.py

```diff
@@ -217,13 +226,24 @@ def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
             self.ds.commit(f"deleted {len(ids)} samples", allow_empty=True)
             self.ds.summary()
 
-    def query(self, query: VectorStoreQuery) -> VectorStoreQueryResult:
+    def query(
+        self, query: VectorStoreQuery, filter: Optional[str] = None
+    ) -> VectorStoreQueryResult:
         """Query index for top k most similar nodes.
 
         Args:
             query_embedding (List[float]): query embedding
             similarity_top_k (int): top k most similar nodes
         """
+        if filter:
+            if isinstance(filter, dict):
+                filter = partial(dp_filter, filter=filter)
+
+            view = self.ds.filter(filter)
+            return DeepLakeVectorStore(
+                deeplake_dataset=view, token=self.token, read_only=self.read_only
+            )
+
```

Comment:
Did you mean to do the filtering at the constructor level? Or alternatively if the filtering is lightweight, you should just use the filtered view to get the embeddings in this function?

Reply:
yup, I was trying to pass already loaded dataset into DeepLakeVectorStore, in this case it would be filtered dataset. The problem that I was trying to resolve is to do metadata filtering on a query level, if there is a better way to do that, I am happy to discuss